conans/client/build/cmake_flags.py

```diff
@@ -294,9 +295,12 @@ def get_definitions(self):
         definitions.update(build_type_definition(self._forced_build_type, build_type,
                                                  self._generator, self._output))
 
-        if str(os_) == "Macos":
-            if arch == "x86":
-                definitions["CMAKE_OSX_ARCHITECTURES"] = "i386"
+        if tools.is_apple_os(os_):
+            definitions["CMAKE_OSX_ARCHITECTURES"] = tools.to_apple_arch(arch)
+            # xcrun is only available on macOS, otherwise it's cross-compiling and it needs to be
+            # set within CMake toolchain
+            if platform.system == "Darwin":
+                definitions["CMAKE_OSX_SYSROOT"] = tools.XCRun(self._conanfile.settings).sdk_path
```

Comment:
The default CMake behavior is: 

> If not set explicitly the value is initialized by the SDKROOT environment variable, if set, and otherwise computed based on the CMAKE_OSX_DEPLOYMENT_TARGET or the host platform.

Is it expected that conan overrides the SDKROOT env variable if defined? Is the ``xcrun`` using the SDKROOT env var?

Reply:
yes, it does (https://www.manpagez.com/man/1/xcrun/):
```
     SDKROOT         Specifies the SDK to use; overriden by the command line
                     -sdk option.
```